### PR TITLE
Fix Coverall

### PR DESCRIPTION
**Pull request by solancer**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -38,13 +38,12 @@ jobs:
       - name: "Run tox targets for ${{ matrix.python-version }}"
         run: "python -m tox"
 
-      # - name: "Report to coveralls"
+      - name: "Report to coveralls"
       #   # coverage is only created in the py39 environment
       #   # --service=github is a workaround for bug
       #   # https://github.com/coveralls-clients/coveralls-python/issues/251
-      #   if: "matrix.python-version == '3.9'"
```
Comment (jugmac00) [inline .github/workflows/main.yml:45]: coverage data is only generated in the 3.9 environment

Comment (solancer) [reply .github/workflows/main.yml:45]: got it

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -28,23 +28,26 @@ jobs:
...
+
       - name: "Install dependencies"
         run: |
           set -xe
           python -VV
           python -m site
           python -m pip install --upgrade pip setuptools wheel
           python -m pip install --upgrade virtualenv tox tox-gh-actions
+          pip install pytest
+          pip install coveralls
```
Comment (jugmac00) [inline .github/workflows/main.yml:40]: `pytest-cov` is missing.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -28,23 +28,23 @@ jobs:
...
-      #   run: |
-      #     pip install coveralls
-      #     coveralls --service=github
-      #   env:
-      #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: "Report to coveralls"
+        if: matrix.python-version == 3.9
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: coveralls debug --service=github
```
Comment (jugmac00) [inline .github/workflows/main.yml:50]: @solancer Thanks for trying so patiently!

You activated the debug mode, which means that all output is printed to the console and no attempt is made to send it - this is unfortunately the reason why the build succeeds :-/

See here
https://coveralls-python.readthedocs.io/en/latest/troubleshooting.html

**Comment by jugmac00** [comment]
Thanks a lot for your effort!

I give up. Getting coveralls to work when it does not want is too much :-)

**Pull request closed without merging**